### PR TITLE
Add client graphql mutation to store data on server 

### PR DESCRIPTION
**Pull request by nilsjung**
## for testing
you have to edit the current address in `frontend/client/index.js` to the ip of your local machine. 
also make sure, that the _mongodb_ and _server_ are running and listening on the right port and address.

### CODE REVIEW

File: frontend/reducers/index.js [hunk 1]
```diff
@@ -1,25 +1,50 @@
...
             return state;
     }
 }
+
+function addDataConnectionReducer(state, { payload }) {
+    const elements = payload.dataConnectionInformation;
+
+    return {
+        ...state,
+        data: addElementToArray(elements, state.data),
```
Comment (karsil) [inline frontend/reducers/index.js]: Warning `[Unhandled promise rejection: TypeError: Invalid attempt to spread non-iterable instance]`

File: frontend/reducers/index.js [hunk 2]
```diff
@@ -1,25 +1,50 @@
...
+    };
+}
+
+function addElementToArray(element, array) {
+    let elements = [...array];
```
Comment (karsil) [inline frontend/reducers/index.js]: Warning, see above

**Review by karsil — CHANGES_REQUESTED** [review]

**Comment by nilsjung** [comment]
* I fixed the warnings and errors. This was a problem regarding the return value of the `Array.push()` function. This returns the new array length and not the new created array.
* I fixed the different graphql results that were stored local in the data-array. now they are consistent and can be stored safely with `store.data.push`
* I merge the actual master

**Review by karsil — DISMISSED** [review]

**Review by karsil — APPROVED** [review]
Approved with disclaimer to correspondig bug ticket (issue #22)

**Pull request merged**